app/src/main/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpass.kt

```diff
@@ -168,10 +171,79 @@ class HikeRoutesRepositoryOverpass(val client: OkHttpClient) : HikeRoutesReposit
             }
             elementReader.endArray()
           }
+          "tags" -> {
+            elementReader.beginObject() // We're in the tags object of the element
+            parseTags(elementReader).let { pair ->
+              elementName = pair.first
+              description = pair.second
+            }
+            elementReader.endObject()
+          }
           else -> elementReader.skipValue()
         }
       }
-      return HikeRoute(id, boundsBuilder.build(), points)
+      return HikeRoute(id, boundsBuilder.build(), points, elementName, description)
+    }
+
+    /**
+     * Parses the tags from the Overpass API response. The reader is supposed to already be in the
+     * tags object.
+     *
+     * @param tagsReader The reader for the tags object.
+     * @return The name and description of the route, in a Pair, the name first, the description
+     *   after.
+     */
+    private fun parseTags(tagsReader: JsonReader): Pair<String?, String?> {
+      // The name of the route can be in multiple tags, we'll try to get the most relevant one
+      // The usual tag is the name tag, but if it's not present, we'll try the name:fr tag, then
+      // alternative tags...
+      // Since OSM is a public database, we can't be sure of the tags used, so we'll try to get the
+      // most relevant one
+      var name: String? = null
+      var description: String? = null
+      var nameFr: String? = null
+      var otherName: String? = null
+      var from: String? = null
+      var to: String? = null
+      while (tagsReader.hasNext()) {
+        when (tagsReader.nextName()) {
+          // int_name is used for international names
+          "int_name" -> name = tagsReader.nextString()
+          // int_name has priority over name
+          "name" -> if (name == null) name = tagsReader.nextString() else tagsReader.skipValue()
+          "name:fr" -> nameFr = tagsReader.nextString()
```

Comment:
Question: Why do you want a french name here ?

Reply:
Because it's not supposed to be french! I'll change that